### PR TITLE
Backports the new zombification mechanics from the Konyang zombie event.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
- Zombie limbs are now ACTUALLY easier to delimb. There was a longstanding bug that made their maim bonus not apply.
- Added new abilities to every zombie type. Ported the Consume verb from bay, which zombies can use to refresh their HP, gib dead enemies, or simply kill alive ones and rip their ribcage open. Yes, it visibly rips their ribcage open.
- The zombie chem is now called Hylemnomil-Zeta and has Lore:tm:.
- Added the antibody extractor. The only way to get Hylemnomil-Zeta antibodies.
- Added barricade stacks.
- Reworked how the infection works. It starts from an appendage and progresses to the brain. An infection can be stopped by cutting off the infected limb.
- You cannot remove zombie parasites surgically anymore.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm [hunk 1]
```diff
@@ -192,16 +193,16 @@
...
-	slowdown = 2
+	slowdown = 1
 
 	vision_flags = DEFAULT_SIGHT
 
 	reagent_tag = IS_UNDEAD
 
 	rarity_value = 10
-	blurb = "Once a living person, this unholy creature was created either by the power of science or necromancy."
+	blurb = "Once a living person, this unholy creature was created either by the power of science."
```
Comment (GMR25) [inline code/modules/mob/living/carbon/human/species/outsider/undead.dm:205]:
```suggestion
	blurb = "Once a living person, this unholy creature was created by the power of science."
```

File: code/game/machinery/antibody.dm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	return ..()
+
+/obj/machinery/antibody_extractor/attack_hand(mob/user)
+	if(working)
+		to_chat(user, SPAN_WARNING("\The [src] is totally unresponsive while processing the cure!"))
+		return
+
+	if(user == occupant)
+		to_chat(user, SPAN_WARNING("The buckles stop you from doing much at all."))
+		. = ..()
```
Comment (Geevies) [inline code/game/machinery/antibody.dm:34]: you're assigning the parent return value to . here, and then returning FALSE anyway. ..() would work here, or you could just remove this line depending on what attack_hand should do

**Review by Geevies — APPROVED** [review]
this ROCKS

File: code/game/objects/random/loot.dm [hunk 3]
```diff
@@ -406,3 +406,210 @@
...
+	M.update_icon()
+
+/obj/random/gun_with_ammo
+	has_postspawn = TRUE
+
+/obj/random/gun_with_ammo/post_spawn(obj/thing)
+	var/obj/item/gun/projectile/G = thing
+	if(G.ammo_magazine)
+		var/mags = rand(0, 2)
+		if(mags)
```
Comment (FluffyGhoster) [inline code/game/objects/random/loot.dm:527]: This is useless, if mags rolls 0, the for loop will be skipped already

File: code/game/objects/random/loot.dm [hunk 4]
```diff
@@ -406,3 +406,210 @@
...
+				var/obj/item/ammo_magazine/AG = new G.ammo_magazine.type(get_turf(G))
+				if(prob(50))
+					var/to_delete = rand(1, length(AG.stored_ammo))
+					var/delete_counter = 0
+					for(var/obj/item/ammo_casing/AC in AG.stored_ammo)
+						qdel(AC)
+						delete_counter++
+						if(delete_counter > to_delete)
+							break
+						AG.update_icon()
```
Comment (FluffyGhoster) [inline code/game/objects/random/loot.dm:538]:
```suggestion
					AG.update_icon()
```
Single icon update

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm [hunk 5]
```diff
@@ -406,13 +427,185 @@
...
+			target = victims[1]
+		else
+			target = tgui_input_list(src, "Who would you like to consume?", "Zombie", victims)
+	else //NPCs
+		if (length(victims) > 0)
+			target = victims[1]
+
+	if (!target)
+		to_chat(src, SPAN_WARNING("You aren't on top of a victim!"))
+		return
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/species/outsider/undead.dm:487]: Seems redundant with what is below? Or at least, the chat message doesn't reflect the failure reason correctly?

File: code/modules/organs/subtypes/parasite/black_kois.dm [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+	origin_tech = list(TECH_BIO = 7)
+
+	egg = /singleton/reagent/kois/black
+
+/obj/item/organ/internal/parasite/blackkois/process()
+	..()
+
+	if(prob(10) && (owner.can_feel_pain()))
+		if(stage < 3)
+			to_chat(owner, "<span class='warning'>You feel a stinging pain in your head!</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/black_kois.dm:26]:
```suggestion
			to_chat(owner, SPAN_WARNING("You feel a stinging pain in your head!"))
```

File: code/modules/organs/subtypes/parasite/black_kois.dm [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	if(stage >= 2)
+		if(prob(10) && !(owner.species.flags & NO_BREATHE))
+			owner.visible_message("<b>[owner]</b> gasps for air!")
+			owner.losebreath += 5
+
+	if(stage >= 3)
+		set_light(-1.5, 6, "#FFFFFF")
+		if(!(GLOB.all_languages[LANGUAGE_VAURCA] in owner.languages))
+			owner.add_language(LANGUAGE_VAURCA)
+			to_chat(owner, "<span class='notice'> Your mind expands, and your thoughts join the unity of the Hivenet.</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/black_kois.dm:41]:
```suggestion
			to_chat(owner, SPAN_NOTICE("Your mind expands, and your thoughts join the unity of the Hivenet."))
```

File: code/modules/organs/subtypes/parasite/black_kois.dm [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+			owner.reagents.add_reagent(/singleton/reagent/kois/black, 4)
+
+		else if(prob(5))
+			if(stage < 4)
+				to_chat(owner, SPAN_GOOD("In your struggle, a part of you wishes for the spread to continue."))
+			else
+				to_chat(owner, SPAN_GOOD("You cannot ever believe that you struggled against this feeling. You are home. You are happy."))
+
+		else if(prob(10))
+			to_chat(owner, "<span class='warning'>You feel disorientated!</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/black_kois.dm:57]:
```suggestion
			to_chat(owner, SPAN_WARNING("You feel disorientated!"))
```

File: code/modules/organs/subtypes/parasite/black_kois.dm [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+			B.max_damage = B.max_damage*0.45 //yeah your brain is severely messed up man
+		if(5)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] wildly, [target.get_pronoun("his")] brain too damaged to function. A horrible choking sound emerges from [target.get_pronoun("his")] mouth, before it is mercifully silenced as [target.get_pronoun("he")] falls still."),\
+			SPAN_DANGER("You feel the cold of the knife cutting and cutting, but it is too late. There is nothing left of you but the Lii'dra's will, now. A final cut - and then you are granted the only escape that still remains to you."))
+			to_chat(target, SPAN_GOOD("Your function is fulfilled. We have no further need of you. Releasing control."))
+			target.death(FALSE) //too late, there is no brain left to save
+	if(GLOB.all_languages[LANGUAGE_VAURCA] in target.languages && stage >= 3 && !isvaurca(target))
+		target.remove_language(LANGUAGE_VAURCA)
+		to_chat(target, "<span class='warning'>Your mind suddenly grows dark as the unity of the Hive is torn from you.</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/black_kois.dm:163]:
```suggestion
		to_chat(target, SPAN_WARNING("Your mind suddenly grows dark as the unity of the Hive is torn from you."))
```

File: code/modules/organs/subtypes/parasite/kois.dm [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+	egg = /singleton/reagent/kois
+
+/obj/item/organ/internal/parasite/kois/process()
+	..()
+
+	if (!owner)
+		return
+
+	if(prob(10) && (owner.can_feel_pain()))
+		to_chat(owner, "<span class='warning'>You feel a stinging pain in your abdomen!</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/kois.dm:24]:
```suggestion
		to_chat(owner, SPAN_WARNING("You feel a stinging pain in your abdomen!"))
```

File: code/modules/organs/subtypes/parasite/kois.dm [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+	if(stage >= 2)
+		if(prob(10) && !(owner.species.flags & NO_BREATHE))
+			owner.visible_message("<b>[owner]</b> gasps for air!")
+			owner.losebreath += 5
+
+	if(stage >= 3)
+		set_light(1, l_color = "#E6E600")
+		if(prob(10))
+			to_chat(owner, "<span class='warning'>You feel something squirming inside of you!</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/kois.dm:43]:
```suggestion
			to_chat(owner, SPAN_WARNING("You feel something squirming inside of you!"))
```

File: code/modules/organs/subtypes/parasite/kois.dm [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+			owner.reagents.add_reagent(/singleton/reagent/toxin/phoron, 8)
+			owner.reagents.add_reagent(/singleton/reagent/kois, 5)
+
+	if(stage >= 4)
+		if(prob(10))
+			to_chat(owner, "<span class='danger'>You feel something alien coming up your throat!</span>")
```
Comment (FluffyGhoster) [inline code/modules/organs/subtypes/parasite/kois.dm:49]:
```suggestion
			to_chat(owner, SPAN_DANGER("You feel something alien coming up your throat!"))
```

**Review by FluffyGhoster — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**